js_modules/dagster-ui/packages/ui-core/src/sensors/SensorSwitch.tsx

```diff
@@ -61,6 +91,74 @@ export const SensorSwitch = (props: Props) => {
 
   const running = status === InstigationStatus.RUNNING;
 
+  if (
+    !running &&
+    canStartSensor &&
+    sensor.sensorType === SensorType.RUN_STATUS &&
+    _isRunStatusSensorLagging(cursor)
+  ) {
+    // We are turning back on a sensor that has a persisted cursor for a position far back in time.
+    // We should warn the user that this is what is happening.
+    const last_processed_timestamp = _parseRunStatusSensorCursor(cursor);
+    return (
+      <>
+        <Checkbox
+          format="switch"
+          disabled={toggleOnInFlight || toggleOffInFlight}
+          checked={running || toggleOnInFlight}
+          onChange={() => setShowRunStatusWarningDialog(true)}
```

Comment:
It looks like the dialog is assuming that the sensor is being turned on, but this looks like the dialog will also be shown when the sensor is being turned off. Is that accurate?